context/BundleData/BundleDataProvider.tsx

```diff
@@ -0,0 +1,182 @@
+import { useRouter } from "next/router"
+import * as React from "react"
+import { useCallback, useEffect } from "react"
+import { useState } from "react"
+
+type BundleDataContextProps = {
+  children: React.ReactNode | React.ReactNode[]
+}
+
+// If we want to display that theres more pages, we'll need to check on ther requests
+// const PAGES_AHEAD = 5
+
+// TODO Update
+export type Block = {
+  hash: string
+  block_number: string
+  miner_reward?: number
+  gas_used?: number
+  coinbase_transfers: number
+  gas_price: number
+  miner: string
+  // TODO: type this 
+  transactions: any[]
+}
+
+
+export type Transaction = {
+  block_number: number
+  bundle_index: number
+  coinbase_transfer: number
+  eoa_address: string
+  gas_price: number
+  gas_used: number
+  to_address: string
+  total_miner_reward: number
+  transaction_hash: string
+  tx_index: number
+}
+
+export interface IBundleFilters {
+  from?: string | string[]
+  to?: string
+  block_number?: string
+  limit: number
+}
+
+interface IBundleDataContext {
+  blocks: Block[]
+  page: number
+  setPage: (page: number) => void
+  morePages: boolean
+  filters: IBundleFilters
+  setFilters: (newFilter: IBundleFilters) => void
+}
+
+const BundleDataContext = React.createContext<IBundleDataContext | undefined>(undefined)
+
+const BundleDataProvider = ({ children }: BundleDataContextProps) => {
+  const { query } = useRouter()
+  const [blocks, setBlocks] = useState<Block[]>([])
+  const [page, setPage] = useState<number>(1)
+  const [morePages, setMorePages] = useState<boolean>(false)
+  const [filters, _setFilters] = useState<IBundleFilters>({
+    limit: 10
+  })
+  const [pageMutex, setPageMutex] = useState(false)
```

Comment:
what do you think of naming this state var `loading`/`isLoading`? I think it makes much easier to understand its purpose and how to use it

```
if (loading) {
  ...
```

Reply:
Personally I keep `loading` type that is specifically if its exposed to the UI, in this case the mutex is specific to page change actions